### ISSUE TITLE
Fails to compile - no method named `to_lowercase` found for reference `&str` in the current scope

### ISSUE THREAD

**Issue by JonathanWilbur** [mikelodder7/celes#14]
I get this error, among other similar ones:

```
no method named `to_lowercase` found for reference `&str` in the current scope
```

```
rustc --version
rustc 1.71.0 (8ede3aae2 2023-07-12)
cargo --version
cargo 1.71.0 (cfd3bbd8f 2023-06-08)
```

This is my `Cargo.toml`:

```toml
[package]
name = "x500"
version = "0.1.0"
edition = "2021"
rust-version = "1.65"
authors = [
    "Jonathan M. Wilbur <jonathan@wilbur.space>",
    "Wildboar Software <jonathan.wilbur@wildboarsoftware.com>",
]
description = "X.500 directory data structures and functionality"
readme = "README.md"
repository = "https://github.com/JonathanWilbur/asn1.rs/tree/master/x500"
license = "MIT"
keywords = ["x500", "directory", "ldap"]
categories = ["network-programming"]

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
asn1 = { path = "../asn1" }
x690 = { path = "../x690" }
ldap = { path = "../ldap" }
teletex = { path = "../teletex" }
nsap-address = { path = "../nsap-address" }
hex = "0.4.3"
punycode = "0.4.1"
cow-utils = "0.1.3"
celes = "2.4.0"

```

**Comment by JonathanWilbur** [comment]
```
error[E0599]: no method named `join` found for array `[&'static str; 1]` in the current scope
   --> /home/jonathan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/celes-2.4.0/src/tables.rs:117:42
    |
117 |                 write!(f, "[{}]", self.0.join(","))
    |                                          ^^^^ method not found in `[&str; 1]`
...
238 | lookup!(MarshallIslandsTable, MarshallIslands, "The Marshall Islands", 1, "MarshallIslands" => "marshallislands");
    | ----------------------------------------------------------------------------------------------------------------- in this macro invocation
    |
    = note: this error originates in the macro `lookup` (in Nightly builds, run with -Z macro-backtrace for more info)
```

**Comment by mikelodder7** [comment]
About time I fixed this. Resolved in new release `2.5.0`
